chain/test_chain.go

```diff
@@ -148,19 +159,24 @@ func NewTestChain(genesisAlloc core.GenesisAlloc, testChainConfig *config.TestCh
 	}
 
 	// Create an in-memory database
-	keyValueStore := memorydb.New()
-	db := rawdb.NewDatabase(keyValueStore)
+	db := rawdb.NewMemoryDatabase()
+	config := &triedb.Config{
+		HashDB: hashdb.Defaults,
+		// PathDB: pathdb.Defaults,
+	}
+	triedb := triedb.NewDatabase(db, config)
+	// TODO	CleanCacheSize: 256,
+	// this is important in keeping the database performant, so it does not need to fetch repetitively.
+	// https://github.com/ethereum/go-ethereum/issues/30099
```

Comment:
I defaulted to use HashDB for now but I am not sure if we should use the catch given we are not committing to triedb any longer and the comments in https://github.com/ethereum/go-ethereum/issues/30099